src/main/groovy/edu/oregonstate/mist/osuevents/db/CacheDAO.groovy

```diff
@@ -87,23 +91,33 @@ class CacheDAO {
         places
     }
 
+    /**
+     * Use as an intermediary method for vendor API endpoints that require pagination.
+     * Iterates through results until all pages have been requested and collected.
+     */
     private def pageIteration(String resource) {
         def data = []
-        Integer page = 1
-        def query = [pp:100]
-        query['page'] = page
+        def query = [pp:100, page:1]
 
-        data.add(page - 1,
+        data.add(query['page'] - 1,
                 jsonSlurper.parseText(
                         sendRequest(resource,
                                 query)
                 )
         )
 
-        while (data[page - 1].page.current != data[page - 1].page.total) {
-            page++
-            query['page'] = page
-            data.add(page - 1,
+        Integer expectedPages = data[query['page'] - 1].page.total
+
+        pageLoop:
+        while (data[query['page'] - 1].page.current < data[query['page'] - 1].page.total) {
+            query['page']++
+
+            if (expectedPages < query['page']) {
+                println("breaking")
```

Comment:
If you get to this line, there was an unexpected bug in pagination. Why not use log.error or log.info to point out this instead of println?